artist.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"context"
+	"log"
+
+	"github.com/zmb3/spotify"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+type artist struct {
+	id         string
+	name       string
+	popularity int
+}
+
+func getByTrackId(id string) artist {
+
+	authConfig := &clientcredentials.Config{
+		ClientID:     "4779b9533e004287b6536fd8c5325adf",
+		ClientSecret: "8dbf0a481f8b4de0901fbc9661f6036c",
+		TokenURL:     spotify.TokenURL,
+	}
+
+	accessToken, err := authConfig.Token(context.Background())
+	if err != nil {
+		log.Fatalf("error retrieving access token: %v", err)
+	}
+
+	client := spotify.Authenticator{}.NewClient(accessToken)
```

Comment:
Each of your "services" (files) is handling authenticating with spotify by themselves. Probably better to have a dedicated class for dealing with authentication, which can then be provided to all the services that need it

Reply:
Actually, just remembered go has [Contexts](https://www.digitalocean.com/community/tutorials/how-to-use-contexts-in-go) for stuff like this (you're even using `context.Background()` already)

Reply:
have moved auth into a separate service